<!DOCTYPE html>
<html lang='en'>
<head>
<title>0003-various-security-fixes.patch - rpms/groff.git - groff</title>
<meta name='generator' content='cgit v1.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit-data/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='http://src.fedoraproject.org/cgit/rpms/groff.git/atom/0003-various-security-fixes.patch?h=master' type='application/atom+xml'/>
<link rel='vcs-git' href='https://src.fedoraproject.org/git/rpms/groff.git' title='rpms/groff.git Git repository'/>
</head>
<body>
<script src="https://apps.fedoraproject.org/fedmenu/js/jquery-1.11.2.min.js"></script>
<script src="https://apps.fedoraproject.org/fedmenu/js/fedmenu.js"></script>

<script>
    var base = 'https://apps.fedoraproject.org/';
    var tokens = window.location.href.split('/');
  var repo = null;
  if (tokens.length > 5 && tokens[4] == 'rpms' && tokens[5] != '') {
    repo = tokens[5].split('.')[0];
  }
  fedmenu({
      'url': base + 'js/data.js',
      'mimeType': 'application/javascript',
      'position': 'bottom-right',
      'package': repo,
  });
</script>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/cgit/'><img src='/cgit-data/cgit.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/cgit/'>index</a> : <a title='rpms/groff.git' href='/cgit/rpms/groff.git/'>rpms/groff.git</a></td><td class='form'><form method='get'>
<select name='h' onchange='this.form.submit();'>
<option value='f10'>f10</option>
<option value='f11'>f11</option>
<option value='f12'>f12</option>
<option value='f13'>f13</option>
<option value='f14'>f14</option>
<option value='f15'>f15</option>
<option value='f16'>f16</option>
<option value='f17'>f17</option>
<option value='f18'>f18</option>
<option value='f19'>f19</option>
<option value='f20'>f20</option>
<option value='f21'>f21</option>
<option value='f22'>f22</option>
<option value='f23'>f23</option>
<option value='f24'>f24</option>
<option value='f25'>f25</option>
<option value='f26'>f26</option>
<option value='f27'>f27</option>
<option value='f28'>f28</option>
<option value='f7'>f7</option>
<option value='f8'>f8</option>
<option value='f9'>f9</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' value='switch'/></form></td></tr>
<tr><td class='sub'>groff</td><td class='sub right'>Jesse Keating</td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/rpms/groff.git/'>summary</a><a href='/cgit/rpms/groff.git/refs/'>refs</a><a href='/cgit/rpms/groff.git/log/0003-various-security-fixes.patch'>log</a><a class='active' href='/cgit/rpms/groff.git/tree/0003-various-security-fixes.patch'>tree</a><a href='/cgit/rpms/groff.git/commit/0003-various-security-fixes.patch'>commit</a><a href='/cgit/rpms/groff.git/diff/0003-various-security-fixes.patch'>diff</a><a href='/cgit/rpms/groff.git/stats/0003-various-security-fixes.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/rpms/groff.git/log/0003-various-security-fixes.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/rpms/groff.git/tree/'>root</a>/<a href='/cgit/rpms/groff.git/tree/0003-various-security-fixes.patch'>0003-various-security-fixes.patch</a></div><div class='content'>blob: 7dd1beaeaaa64934b14009dd2900fb37274775ce (<a href='/cgit/rpms/groff.git/plain/0003-various-security-fixes.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
<a id='n106' href='#n106'>106</a>
<a id='n107' href='#n107'>107</a>
<a id='n108' href='#n108'>108</a>
<a id='n109' href='#n109'>109</a>
<a id='n110' href='#n110'>110</a>
<a id='n111' href='#n111'>111</a>
<a id='n112' href='#n112'>112</a>
<a id='n113' href='#n113'>113</a>
<a id='n114' href='#n114'>114</a>
<a id='n115' href='#n115'>115</a>
<a id='n116' href='#n116'>116</a>
<a id='n117' href='#n117'>117</a>
<a id='n118' href='#n118'>118</a>
<a id='n119' href='#n119'>119</a>
<a id='n120' href='#n120'>120</a>
<a id='n121' href='#n121'>121</a>
<a id='n122' href='#n122'>122</a>
<a id='n123' href='#n123'>123</a>
<a id='n124' href='#n124'>124</a>
<a id='n125' href='#n125'>125</a>
<a id='n126' href='#n126'>126</a>
<a id='n127' href='#n127'>127</a>
<a id='n128' href='#n128'>128</a>
<a id='n129' href='#n129'>129</a>
<a id='n130' href='#n130'>130</a>
<a id='n131' href='#n131'>131</a>
<a id='n132' href='#n132'>132</a>
<a id='n133' href='#n133'>133</a>
<a id='n134' href='#n134'>134</a>
<a id='n135' href='#n135'>135</a>
<a id='n136' href='#n136'>136</a>
<a id='n137' href='#n137'>137</a>
<a id='n138' href='#n138'>138</a>
<a id='n139' href='#n139'>139</a>
<a id='n140' href='#n140'>140</a>
<a id='n141' href='#n141'>141</a>
<a id='n142' href='#n142'>142</a>
<a id='n143' href='#n143'>143</a>
<a id='n144' href='#n144'>144</a>
<a id='n145' href='#n145'>145</a>
<a id='n146' href='#n146'>146</a>
<a id='n147' href='#n147'>147</a>
<a id='n148' href='#n148'>148</a>
<a id='n149' href='#n149'>149</a>
<a id='n150' href='#n150'>150</a>
<a id='n151' href='#n151'>151</a>
<a id='n152' href='#n152'>152</a>
<a id='n153' href='#n153'>153</a>
<a id='n154' href='#n154'>154</a>
<a id='n155' href='#n155'>155</a>
<a id='n156' href='#n156'>156</a>
<a id='n157' href='#n157'>157</a>
<a id='n158' href='#n158'>158</a>
<a id='n159' href='#n159'>159</a>
<a id='n160' href='#n160'>160</a>
<a id='n161' href='#n161'>161</a>
<a id='n162' href='#n162'>162</a>
<a id='n163' href='#n163'>163</a>
<a id='n164' href='#n164'>164</a>
<a id='n165' href='#n165'>165</a>
<a id='n166' href='#n166'>166</a>
<a id='n167' href='#n167'>167</a>
<a id='n168' href='#n168'>168</a>
<a id='n169' href='#n169'>169</a>
<a id='n170' href='#n170'>170</a>
<a id='n171' href='#n171'>171</a>
<a id='n172' href='#n172'>172</a>
<a id='n173' href='#n173'>173</a>
<a id='n174' href='#n174'>174</a>
<a id='n175' href='#n175'>175</a>
<a id='n176' href='#n176'>176</a>
<a id='n177' href='#n177'>177</a>
<a id='n178' href='#n178'>178</a>
<a id='n179' href='#n179'>179</a>
<a id='n180' href='#n180'>180</a>
<a id='n181' href='#n181'>181</a>
<a id='n182' href='#n182'>182</a>
<a id='n183' href='#n183'>183</a>
<a id='n184' href='#n184'>184</a>
<a id='n185' href='#n185'>185</a>
<a id='n186' href='#n186'>186</a>
<a id='n187' href='#n187'>187</a>
<a id='n188' href='#n188'>188</a>
<a id='n189' href='#n189'>189</a>
<a id='n190' href='#n190'>190</a>
<a id='n191' href='#n191'>191</a>
<a id='n192' href='#n192'>192</a>
<a id='n193' href='#n193'>193</a>
<a id='n194' href='#n194'>194</a>
<a id='n195' href='#n195'>195</a>
<a id='n196' href='#n196'>196</a>
<a id='n197' href='#n197'>197</a>
<a id='n198' href='#n198'>198</a>
<a id='n199' href='#n199'>199</a>
<a id='n200' href='#n200'>200</a>
<a id='n201' href='#n201'>201</a>
<a id='n202' href='#n202'>202</a>
<a id='n203' href='#n203'>203</a>
<a id='n204' href='#n204'>204</a>
<a id='n205' href='#n205'>205</a>
<a id='n206' href='#n206'>206</a>
<a id='n207' href='#n207'>207</a>
<a id='n208' href='#n208'>208</a>
<a id='n209' href='#n209'>209</a>
<a id='n210' href='#n210'>210</a>
<a id='n211' href='#n211'>211</a>
<a id='n212' href='#n212'>212</a>
<a id='n213' href='#n213'>213</a>
<a id='n214' href='#n214'>214</a>
<a id='n215' href='#n215'>215</a>
<a id='n216' href='#n216'>216</a>
<a id='n217' href='#n217'>217</a>
<a id='n218' href='#n218'>218</a>
<a id='n219' href='#n219'>219</a>
<a id='n220' href='#n220'>220</a>
<a id='n221' href='#n221'>221</a>
<a id='n222' href='#n222'>222</a>
<a id='n223' href='#n223'>223</a>
<a id='n224' href='#n224'>224</a>
<a id='n225' href='#n225'>225</a>
<a id='n226' href='#n226'>226</a>
<a id='n227' href='#n227'>227</a>
<a id='n228' href='#n228'>228</a>
</pre></td>
<td class='lines'><pre><code>From 36115e102859badb08cb5b2398de6b0ba45421d3 Mon Sep 17 00:00:00 2001
From: Jan Vcelak &lt;jvcelak&#64;redhat.com&gt;
Date: Tue, 4 Nov 2014 14:36:47 +0100
Subject: [PATCH] various security fixes

CVE-2009-5044 (#709413)
CVE-2009-5080 (#720058)
CVE-2009-5081 (#720057)

Based on: http://cvsweb.openwall.com/cgi/cvsweb.cgi/Owl/packages/groff/groff-1.20.1-owl-tmp.diff?rev=1.2;content-type=text%2Fplain
Resolves: #709415, #720060
Signed-off-by: Jan Vcelak &lt;jvcelak&#64;redhat.com&gt;
<span class="hl kwb">---</span>
 contrib/eqn2graph/eqn2graph.sh    |  2 ++
 contrib/gdiffmk/tests/runtests.in |  5 +++--
 contrib/grap2graph/grap2graph.sh  |  2 ++
 contrib/groffer/main_subs.pl      | 10 +++++-----
 contrib/groffer/roff2.pl          |  2 +-
 contrib/pdfmark/pdfroff.man       |  5 +++--
 contrib/pic2graph/pic2graph.sh    |  2 ++
 doc/fixinfo.sh                    |  5 +++--
 doc/groff.info-2                  |  6 +++---
 doc/groff.texinfo                 |  6 +++---
 gendef.sh                         | 10 +++-------
 11 files changed, 30 insertions(+), 25 deletions(-)

diff --git a/contrib/eqn2graph/eqn2graph.sh b/contrib/eqn2graph/eqn2graph.sh
index ee7cc5f..13edf78 100644
<span class="hl kwb">--- a/contrib/eqn2graph/eqn2graph.sh</span>
<span class="hl kwa">+++ b/contrib/eqn2graph/eqn2graph.sh</span>
&#64;&#64; -67,6 +67,8 &#64;&#64; for d in &quot;$GROFF_TMPDIR&quot; &quot;$TMPDIR&quot; &quot;$TMP&quot; &quot;$TEMP&quot; /tmp; do
 
     tmp=$d/eqn2graph$$-$RANDOM
     (umask 077 &amp;&amp; mkdir $tmp) 2&gt; /dev/null &amp;&amp; break
<span class="hl kwa">+</span>
<span class="hl kwa">+    tmp=</span>
 done;
 if test -z &quot;$tmp&quot;; then
     echo &quot;$0: cannot create temporary directory&quot; &gt;&amp;2
diff --git a/contrib/gdiffmk/tests/runtests.in b/contrib/gdiffmk/tests/runtests.in
index 714ce48..40a35c4 100644
<span class="hl kwb">--- a/contrib/gdiffmk/tests/runtests.in</span>
<span class="hl kwa">+++ b/contrib/gdiffmk/tests/runtests.in</span>
&#64;&#64; -56,8 +56,9 &#64;&#64; function TestResult {
 	fi
 }
 
<span class="hl kwb">-tmpfile=/tmp/$$</span>
<span class="hl kwb">-trap 'rm -f ${tmpfile}' 0 1 2 3 15</span>
<span class="hl kwa">+tmpfile=&quot;`mktemp -t gdiffmk-runtests.XXXXXXXXXX`&quot; || exit</span>
<span class="hl kwa">+trap 'rm -f -- &quot;$tmpfile&quot;' EXIT</span>
<span class="hl kwa">+trap 'trap - EXIT; rm -f -- &quot;$tmpfile&quot;; exit 1' HUP INT QUIT TERM</span>
 
 #	Run tests.
 
diff --git a/contrib/grap2graph/grap2graph.sh b/contrib/grap2graph/grap2graph.sh
index 58544e1..aeab832 100644
<span class="hl kwb">--- a/contrib/grap2graph/grap2graph.sh</span>
<span class="hl kwa">+++ b/contrib/grap2graph/grap2graph.sh</span>
&#64;&#64; -63,6 +63,8 &#64;&#64; for d in &quot;$GROFF_TMPDIR&quot; &quot;$TMPDIR&quot; &quot;$TMP&quot; &quot;$TEMP&quot; /tmp; do
 
     tmp=$d/grap2graph$$-$RANDOM
     (umask 077 &amp;&amp; mkdir $tmp) 2&gt; /dev/null &amp;&amp; break
<span class="hl kwa">+</span>
<span class="hl kwa">+    tmp=</span>
 done;
 if test -z &quot;$tmp&quot;; then
     echo &quot;$0: cannot create temporary directory&quot; &gt;&amp;2
diff --git a/contrib/groffer/main_subs.pl b/contrib/groffer/main_subs.pl
index 90627cc..76896cd 100644
<span class="hl kwb">--- a/contrib/groffer/main_subs.pl</span>
<span class="hl kwa">+++ b/contrib/groffer/main_subs.pl</span>
&#64;&#64; -1239,7 +1239,7 &#64;&#64; sub main_temp {
   our $fh_stdin;
   our $tmp_cat;
   our $tmp_stdin;
<span class="hl kwb">-  my $template = 'groffer_' . &quot;$$&quot; . '_XXXX';</span>
<span class="hl kwa">+  my $template = 'groffer_' . &quot;$$&quot; . '_XXXXXXXXXX';</span>
   foreach ($ENV{'GROFF_TMPDIR'}, $ENV{'TMPDIR'}, $ENV{'TMP'}, $ENV{'TEMP'},
 	   $ENV{'TEMPDIR'}, File::Spec-&gt;catfile($ENV{'HOME'}, 'tmp')) {
     if ($_ &amp;&amp; -d $_ &amp;&amp; -w $_) {
&#64;&#64; -1271,12 +1271,12 &#64;&#64; sub main_temp {
 
   # further argument: SUFFIX =&gt; '.sh'
   if ($Debug{'KEEP'}) {
<span class="hl kwb">-    ($fh_cat, $tmp_cat) = tempfile(',cat_XXXX', DIR =&gt; $tmpdir);</span>
<span class="hl kwb">-    ($fh_stdin, $tmp_stdin) = tempfile(',stdin_XXXX', DIR =&gt; $tmpdir);</span>
<span class="hl kwa">+    ($fh_cat, $tmp_cat) = tempfile(',cat_XXXXXXXXXX', DIR =&gt; $tmpdir);</span>
<span class="hl kwa">+    ($fh_stdin, $tmp_stdin) = tempfile(',stdin_XXXXXXXXXX', DIR =&gt; $tmpdir);</span>
   } else {
<span class="hl kwb">-    ($fh_cat, $tmp_cat) = tempfile(',cat_XXXX', UNLINK =&gt; 1,</span>
<span class="hl kwa">+    ($fh_cat, $tmp_cat) = tempfile(',cat_XXXXXXXXXX', UNLINK =&gt; 1,</span>
 				   DIR =&gt; $tmpdir);
<span class="hl kwb">-    ($fh_stdin, $tmp_stdin) = tempfile(',stdin_XXXX', UNLINK =&gt; 1,</span>
<span class="hl kwa">+    ($fh_stdin, $tmp_stdin) = tempfile(',stdin_XXXXXXXXXX', UNLINK =&gt; 1,</span>
 				       DIR =&gt; $tmpdir);
   }
 }				# main_temp()
diff --git a/contrib/groffer/roff2.pl b/contrib/groffer/roff2.pl
index d8dad3f..f0ca9f2 100755
<span class="hl kwb">--- a/contrib/groffer/roff2.pl</span>
<span class="hl kwa">+++ b/contrib/groffer/roff2.pl</span>
&#64;&#64; -123,7 +123,7 &#64;&#64; if ($Has_Groffer) {
 	last;
       }
     }
<span class="hl kwb">-    my $template = $Name . '_XXXX';</span>
<span class="hl kwa">+    my $template = $Name . '_XXXXXXXXXX';</span>
     my ($fh, $stdin);
     if ($tempdir) {
       ($fh, $stdin) = tempfile($template, UNLINK =&gt; 1, DIR =&gt; $tempdir) ||
diff --git a/contrib/pdfmark/pdfroff.man b/contrib/pdfmark/pdfroff.man
index ec412bb..faf2898 100644
<span class="hl kwb">--- a/contrib/pdfmark/pdfroff.man</span>
<span class="hl kwa">+++ b/contrib/pdfmark/pdfroff.man</span>
&#64;&#64; -555,7 +555,7 &#64;&#64; defaults to
 .B GROFF_TMPDIR
 Identifies the directory in which
 .B pdfroff
<span class="hl kwb">-should create temporary files.</span>
<span class="hl kwa">+should create a subdirectory for its temporary files.</span>
 .
 If
 .B \%GROFF_TMPDIR
&#64;&#64; -568,7 +568,8 &#64;&#64; and
 .B TEMP
 are considered in turn, as possible temporary file repositories.
 If none of these are set, then temporary files are created
<span class="hl kwb">-in the current directory.</span>
<span class="hl kwa">+in a subdirectory of</span>
<span class="hl kwa">+.BR /tmp .</span>
 .
 .TP
 .B GROFF_GHOSTSCRIPT_INTERPRETER
diff --git a/contrib/pic2graph/pic2graph.sh b/contrib/pic2graph/pic2graph.sh
index 72c5477..6b3360d 100644
<span class="hl kwb">--- a/contrib/pic2graph/pic2graph.sh</span>
<span class="hl kwa">+++ b/contrib/pic2graph/pic2graph.sh</span>
&#64;&#64; -78,6 +78,8 &#64;&#64; for d in &quot;$GROFF_TMPDIR&quot; &quot;$TMPDIR&quot; &quot;$TMP&quot; &quot;$TEMP&quot; /tmp; do
     tmp=$d/pic2graph$$-$RANDOM
     (umask 077 &amp;&amp; mkdir $tmp) 2&gt; /dev/null \
     &amp;&amp; break
<span class="hl kwa">+</span>
<span class="hl kwa">+    tmp=</span>
 done;
 if test -z &quot;$tmp&quot;; then
     echo &quot;$0: cannot create temporary directory&quot; &gt;&amp;2
diff --git a/doc/fixinfo.sh b/doc/fixinfo.sh
index 2c853f8..6954e6a 100644
<span class="hl kwb">--- a/doc/fixinfo.sh</span>
<span class="hl kwa">+++ b/doc/fixinfo.sh</span>
<span class="hl kwd">&#64;&#64; -22,8 +22,9 &#64;&#64;</span>
 # groff.texinfo macro code.  Hopefully, a new texinfo version makes it
 # unnecessary.
 
<span class="hl kwb">-t=${TMPDIR-.}/gro$$.tmp</span>
<span class="hl kwb">-</span>
<span class="hl kwa">+t=&quot;`mktemp -t groff-fixinfo.XXXXXXXXXX`&quot; || exit</span>
<span class="hl kwa">+trap 'rm -f -- &quot;$t&quot;' EXIT</span>
<span class="hl kwa">+trap 'trap - EXIT; rm -f -- &quot;$t&quot;; exit 1' HUP INT QUIT TERM</span>
 cat $1 | sed '
 1 {
   N
diff --git a/doc/groff.info-2 b/doc/groff.info-2
index 7eaae86..e7dab72 100644
<span class="hl kwb">--- a/doc/groff.info-2</span>
<span class="hl kwa">+++ b/doc/groff.info-2</span>
&#64;&#64; -1697,9 +1697,9 &#64;&#64; not there, 'groff' would not know when to stop.
      time into a document:
 
           .sy perl -e 'printf &quot;.nr H %d\\n.nr M %d\\n.nr S %d\\n&quot;,\
<span class="hl kwb">-                       (localtime(time))[2,1,0]' &gt; /tmp/x\n[$$]</span>
<span class="hl kwb">-          .so /tmp/x\n[$$]</span>
<span class="hl kwb">-          .sy rm /tmp/x\n[$$]</span>
<span class="hl kwa">+                       (localtime(time))[2,1,0]' &gt; timefile\n[$$]</span>
<span class="hl kwa">+          .so timefile\n[$$]</span>
<span class="hl kwa">+          .sy rm timefile\n[$$]</span>
           \nH:\nM:\nS
 
      Note that this works by having the 'perl' script (run by 'sy')
diff --git a/doc/groff.texinfo b/doc/groff.texinfo
index 066b527..83684da 100644
<span class="hl kwb">--- a/doc/groff.texinfo</span>
<span class="hl kwa">+++ b/doc/groff.texinfo</span>
&#64;&#64; -13736,9 +13736,9 &#64;&#64; into a document:
 &#64;pindex perl
 &#64;Example
 .sy perl -e 'printf &quot;.nr H %d\\n.nr M %d\\n.nr S %d\\n&quot;,\
<span class="hl kwb">-             (localtime(time))[2,1,0]' &gt; /tmp/x\n[$$]</span>
<span class="hl kwb">-.so /tmp/x\n[$$]</span>
<span class="hl kwb">-.sy rm /tmp/x\n[$$]</span>
<span class="hl kwa">+             (localtime(time))[2,1,0]' &gt; timefile\n[$$]</span>
<span class="hl kwa">++.so timefile\n[$$]</span>
<span class="hl kwa">++.sy rm timefile\n[$$]</span>
 \nH:\nM:\nS
 &#64;endExample
 
diff --git a/gendef.sh b/gendef.sh
index 41a511b..050bcbe 100644
<span class="hl kwb">--- a/gendef.sh</span>
<span class="hl kwa">+++ b/gendef.sh</span>
&#64;&#64; -33,11 +33,9 &#64;&#64; do
 #define $def&quot;
 done
 
<span class="hl kwb">-# Use $TMPDIR if defined.  Default to cwd, for non-Unix systems</span>
<span class="hl kwb">-# which don't have /tmp on each drive (we are going to remove</span>
<span class="hl kwb">-# the file before we exit anyway).  Put the PID in the basename,</span>
<span class="hl kwb">-# since the extension can only hold 3 characters on MS-DOS.</span>
<span class="hl kwb">-t=${TMPDIR-.}/gro$$.tmp</span>
<span class="hl kwa">+t=&quot;`mktemp -t groff-gendef.XXXXXXXXXX`&quot; || exit</span>
<span class="hl kwa">+trap 'rm -f -- &quot;$t&quot;' EXIT</span>
<span class="hl kwa">+trap 'trap - EXIT; rm -f -- &quot;$t&quot;; exit 1' HUP INT QUIT TERM</span>
 
 sed -e 's/=/ /' &gt;$t &lt;&lt;EOF
 $defs
&#64;&#64; -45,8 +43,6 &#64;&#64; EOF
 
 test -r $file &amp;&amp; cmp -s $t $file || cp $t $file
 
<span class="hl kwb">-rm -f $t</span>
<span class="hl kwb">-</span>
 exit 0
 
 # eof
<span class="hl kwb">-- </span>
1.9.3

</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated by <a href='https://git.zx2c4.com/cgit/about/'>cgit v1.1</a> at 2018-03-03 16:33:44 +0000</div>
</div> <!-- id=cgit -->
</body>
</html>
